### PR TITLE
[NETBEANS-7069] Support Nashorn 15.x for JDK >= 15

### PR DESCRIPTION
**Pull request by cz6ace**
Netbeans does support debugging of legacy Nashorn JavaScript engine bundled with OpenJDK < 15.0. It does not support the the most recent Nashorn 15.4 extracted from OpenJDK, that has different package name.

Change allows debugging both legacy Nashorn JavaScript and also extracted Nashorn with new org.openjdk.nashorn package name.

See issue https://github.com/apache/netbeans/issues/7069

### CODE REVIEW

**Comment by cz6ace** [comment]
Hi @matthiasblaesing ,
thank you for comments, I'll check. To be honest, I tried both scenarios with external application and I simply attached via socket to the jvm. I'll try your project.

And yes, problem will be most probably in the JSJavaBreakpointsManager, this is tricky part, because one has to create breakpoints in advance and the debugger instance not available, so I created two sets of breakpoints. I need also put some explanation for potential reader.

File: java/debugger.jpda.js/src/org/netbeans/modules/debugger/jpda/js/StepIntoJSHandler.java [hunk 1]
```diff
@@ -64,7 +64,7 @@
 @LazyActionsManagerListener.Registration(path="netbeans-JPDASession/Java")
 public class StepIntoJSHandler extends LazyActionsManagerListener implements PropertyChangeListener {
     
-    private static final String SCRIPT_ACCESS_CLASS = "jdk.nashorn.internal.runtime.ScriptFunctionData";    // NOI18N
+    private static final String SCRIPT_ACCESS_CLASS = "org.openjdk.nashorn.internal.runtime.ScriptFunctionData";    // NOI18N
```
[the remark this answers is not in the document]
Comment (cz6ace) [reply java/debugger.jpda.js/src/org/netbeans/modules/debugger/jpda/js/StepIntoJSHandler.java:67]: good catch, you are correct, I missed this. Need to apply for both Nashorns too.

**Comment by cz6ace** [comment]
Hi @matthiasblaesing ,

even though I was able to debug the javascript on my side, I also had those weird breakpoints you mentioned in the screenshots  - and in fact, they revealed that I brought some regression(s). I'll work on it before the vacation ends.

Libor

**Comment by cz6ace** [comment]
Hi @matthiasblaesing,

with the last commit I can debug in both cases, I tried
- OpenJDK 11 (commented the Nashorn in the pom.xml)
- OpenJDK 17 with enabled Nashorn in the pom.xml
First, it stops in the javascript, second, it does not stop in the Nashorn sources (your screenshots)

If you could please check on your side again, it will be appreciated.

My results:
JDK 11:
image

JDK 17:
image

**Comment by matthiasblaesing** [comment]
Thank you for the update. I confirm, that it works for me and I think we should get this in. @cz6ace would you please squash this, so that I can invoke a final CI/CD run. The current test failure is caused by CI/CD picking up commit messages, that are obviously not final.

**Comment by cz6ace** [comment]
Good to hear it works on your side too. I did the squash and rebased to the latest master to be sure there's no surprise.
Please execute the CI/CD, thx!

**Review by matthiasblaesing — APPROVED** [review]
Looks sane to me. Lets get this in.

**Comment by 3143948** [comment]
@cz6ace @matthiasblaesing 
Can this be integrated in the vscode Netbeans extension as well?

**Comment by 3143948** [comment]
This works great with `debugger;` line.
It opens a new script called eval.js which has the source code of the remote script and you can then go line by line.
Is it possible to use the local js source file and use a break point in it instead of using the `debugger;` statements?

**Pull request merged**